### PR TITLE
Strip null values from CloudFormation template resource properties

### PR DESCRIPTION
**Pull request by GurmeharS**
Previously, `stripNullPropsFromTemplateResources` stripped null from top-level resource properties in the compiled CFN template.

This PR:
- adds functionality to traverse the properties deeply, and strips null from nested objects within the properties
- allows for users to provide default value as null if they would like the variable to be removed from the template if the initial value is not available (see #10279)

Closes: #10279

### CODE REVIEW

File: lib/plugins/aws/package/lib/stripNullPropsFromTemplateResources.js [hunk 1]
```diff
@@ -1,16 +1,22 @@
 'use strict';
 
+const stripNullPropsFromObj = (obj) => {
+  Object.entries(obj).forEach(([propName, propVal]) => {
+    if (propVal === null) {
+      delete obj[propName];
+    } else if (propVal && typeof propVal === 'object') {
```
Comment (medikoo) [inline lib/plugins/aws/package/lib/stripNullPropsFromTemplateResources.js:7]: We can simplify to `if (typeof === 'object')`

**Review by medikoo — COMMENTED** [review]
@GurmeharS that looks great ! I've just propose one minor simplification, and it's good to go

File: lib/plugins/aws/package/lib/stripNullPropsFromTemplateResources.js (hunk 1 above)
Comment (GurmeharS) [reply lib/plugins/aws/package/lib/stripNullPropsFromTemplateResources.js]: Good catch! Pushed out an update 👍

**Review by medikoo — APPROVED** [review]
Thank you @GurmeharS !

**Pull request merged**